src/Exception/NonCastableTypeException.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JMS\Serializer\Exception;
+
+class NonCastableTypeException extends RuntimeException
```

Comment:
should this class be abstract?

Reply:
why not. 

also children classes could be declared final?

Reply:
that sounds a good idea!

Reply:
fixed